packages/core/src/lifecycle-registry.ts

```diff
@@ -0,0 +1,211 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: @loopback/core
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {Binding, BindingScope, BindingType, Context} from '@loopback/context';
+import {CoreBindings} from './keys';
+import {LifeCycleObserver} from './lifecycle';
+const CoreTags = CoreBindings.Tags;
+import debugFactory = require('debug');
+const debug = debugFactory('loopback:core:lifecycle');
+
+/**
+ * A group of life cycle observers
+ */
+export type LifeCycleObserverGroup = {
+  group: string;
+  bindings: Readonly<Binding<LifeCycleObserver>>[];
+};
+
+export type LifeCycleObserverOptions = {
+  /**
+   * Notify observers of the same group in parallel, default to `true`
+   */
+  parallel?: boolean;
+};
+
+/**
+ *
+ * A Context subclass that supports life cycle
+ *
+ */
+export class LifeCycleObserverRegistry implements LifeCycleObserver {
+  constructor(
+    protected ctx: Context,
+    protected options: LifeCycleObserverOptions = {parallel: true},
+  ) {}
+  /**
+   * Configures the ordering of life cycle observers by group names
+   */
+  protected groupsByOrder: string[] = ['server'];
+
+  setGroupsByOrder(groups: string[]) {
+    this.groupsByOrder = groups || ['server'];
```

Comment:
I have difficulties to comprehend this group ordering concept and how can application & extension users leverage this configuration point. Could you please explain, perhaps in the form of a documentation as part of this pull request?